cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -586,19 +594,32 @@ private ModelCompletenessRequirements modelCompletenessRequirements(Collection<G
    * @param goalNames Goal names (and empty list of names indicates all goals).
    */
   public boolean meetCompletenessRequirements(List<String> goalNames) {
-    Collection<Goal> goals = goalsByPriority(goalNames).values();
+    Collection<Goal> goals = goalsByPriority(goalNames, false).values();
     MetadataClient.ClusterAndGeneration clusterAndGeneration = _loadMonitor.refreshClusterAndGeneration();
     return goals.stream().allMatch(g -> _loadMonitor.meetCompletenessRequirements(
         clusterAndGeneration, g.clusterModelCompletenessRequirements()));
   }
 
   /**
    * Get a goals by priority based on the goal list.
+   *
+   * @param goals A list of goals to optimize.
+   * @param skipHardGoalCheck Whether check and add all hard goals to {@code goals}.
```

Comment:
If we rephrase it as follows, would it be more clear?
`Whether check and add all hard goals to {@code goals}.` -> `True if the provided {@code goals} do not have to contain all hard goals, false otherwise`